docs/sources/flow/reference/config-blocks/remotecfg.md

```diff
@@ -0,0 +1,92 @@
+---
+aliases:
+- /docs/grafana-cloud/agent/flow/reference/config-blocks/remotecfg/
+- /docs/grafana-cloud/monitor-infrastructure/agent/flow/reference/config-blocks/remotecfg/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/reference/config-blocks/remotecfg/
+- /docs/grafana-cloud/send-data/agent/flow/reference/config-blocks/remotecfg/
+canonical: remotecfgs://grafana.com/docs/agent/latest/flow/reference/config-blocks/remotecfg/
+description: Learn about the remotecfg configuration block
+menuTitle: remotecfg
+title: remotecfg block
+---
+
+# remotecfg block
+
+`remotecfg` is an optional configuration block that enables {{< param "PRODUCT_NAME" >}}
+to fetch and load configuration from a remote endpoint.
+`remotecfg` is specified without a label and can only be provided once per
+configuration file.
+
+The [API definition][] for managing and fetching configuration that the
+`remotecfg` block uses is available under the Apache 2.0 license.
+
+[API definition]: https://github.com/grafana/agent-remote-config
+
+## Example
+
+```river
+remotecfg {
+	url = "SERVICE_URL"
+	basic_auth {
+		username      = USERNAME
+		password_file = PASSWORD_FILE
+	}
+
+	id             = constants.hostname
+	metadata       = {"cluster" = "dev-us-central-0", "namespace" = "agent-otlp"}
+	poll_frequency = "10m"
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name             | Type                 | Description                                      | Default     | Required
+-----------------|----------------------|--------------------------------------------------|-------------|---------
+`url`            | `string`             | The address of the API to poll for configuration | `""`        | no
```

Comment:
suggestion: this is marked as not required because the service itself isn't required for the agent to run, right? But for the service to function, a URL does need to be provided. Perhaps there should be a note below the table that the URL _is_ functionally required if the service is to run.

Reply:
Yes; right now services blocks are always implicitly constructed (so there's no way to have an required field).

We can expand on this with a note below.